AzureIoTHub.agent.lib.nut

```diff
@@ -723,6 +724,10 @@ class AzureIoTHub {
                 _options[optName] <- optVal;
             }
 
+            _msgOptions = {
+                "qos" : _options.qos
```

Comment:
can we handles `retain` as well: https://developer.electricimp.com/api/mqtt/mqttclient/createmessage?

Reply:
We can, but Azure provides limited support for this flag. So we decided just to make a note in the main readme about this.